app/src/main/java/com/ataulm/whatsnext/Images.kt

```diff
@@ -1,16 +1,18 @@
 package com.ataulm.whatsnext
 
+import androidx.annotation.Px
+
 data class Images(private val images: List<Image>) {
 
-    fun bestFor(desiredWidth: Int): Image? {
+    fun bestFor(@Px desiredWidth: Int): Image? {
         if (images.isEmpty()) {
             return null
         }
         return images.reduce { currentImage, nextImage ->
             if (bothBiggerThanDesired(currentImage, nextImage, desiredWidth)) {
                 return smallestOf(currentImage, nextImage)
             }
-            return largestOf(currentImage, nextImage)
+            largestOf(currentImage, nextImage)
```

Comment:
this was returning `bestFor` instead of the reducer function in `reduce`, meaning this was only checking the first two images.

The return keyword isn't necessary.